### PR TITLE
Retrieve all parameters within a hierarchy

### PR DESCRIPTION
**Pull request by benhoIIand**
The documentation suggests that this is already functionality that exists. We've recently tried to request nested parameters but found it wasn't working because the `recursive` flag wasn't set.

### CODE REVIEW

File: src/ssm.test.ts [hunk 1]
```diff
@@ -26,15 +25,15 @@ const p = (name: string, value: string) => ({
 
 describe('The simplest possible config object', () => {
   let server: http.Server
-  let requestedPath: Promise<string>
+  let requestedParams: Promise<ISSMQueryOptions>
```
Comment (bobthemighty) [inline src/ssm.test.ts:28]: Can we drop the hungarian notation? `SSMQueryOptions` is a sensible name, we'll hav enone of your Java/C# inflected weirdness round here, thanks.

File: src/ssm.test.ts [hunk 2]
```diff
@@ -50,15 +49,19 @@ describe('The simplest possible config object', () => {
   describe('When we provide a path prefix', () => {
     it('should prepend the prefix to the requested path', async () => {
```
Comment (bobthemighty) [inline src/ssm.test.ts:50]: This test name no longer adequately describes the assertions. Options include:

```
let awsSSMParams;

/// make the call and set the params object
beforeAll() ...

it('should prepend the prefix, () =>{});

it('should pass the recursive flag', () =>{});

it('should request decryption', () =>{});

}
```

or just be a BAD LAZY HUMAN

```
it('should pass the correct params');
```

File: src/ssm.test.ts [hunk 3]
```diff
@@ -71,8 +74,8 @@ describe('The new reader', () => {
     process.env.AWS_REGION = 'eu-west-1'
     process.env.AWS_ACCESS_KEY_ID = 'foo'
     process.env.AWS_SECRET_ACCESS_KEY = 'bar'
-    let resolver: (path: string) => void
-    requestedPath = new Promise((resolve, reject) => {
+    let resolver: (obj: ISSMQueryOptions) => void
```
Comment (bobthemighty) [inline src/ssm.test.ts:77]: `obj` is the worst name for a variable. It's longer than `x` and just as meaningful. `params` ? `options` ?

File: src/ssm.test.ts (hunk 1 above)
Comment (benhoIIand) [reply src/ssm.test.ts:28]: Ahh, the build failed and told me to do it this way, promise

Comment (bobthemighty) [reply src/ssm.test.ts:28]: ewww.... okay... leave it, and I'll fix it up another time. I'll need to remember how the linter is configured.

File: tslint.json [hunk 4]
```diff
@@ -11,6 +11,7 @@
     "arrow-parens": false,
     "curly": [true, "ignore-same-line"],
     "max-line-length": false,
-    "quotemark": [true, "single", "avoid-escape", "avoid-template"]
+    "quotemark": [true, "single", "avoid-escape", "avoid-template"],
+    "interface-name" : [true, "never-prefix"]
```
Comment (bobthemighty) [inline tslint.json:15]: You are a good and kind person @maghrout . Your parents and line manager will be very proud of your efforts. :heart:

Comment (maghrout) [reply tslint.json:15]: I am screenshotting this and making it my wallpaper

**Comment by benhoIIand** [comment]
@maghrout is gunna clean up my mess 😬

**Comment by maghrout** [comment]
> @maghrout is gunna clean up my mess 😬

**Comment by maghrout** [comment]
Not unless I accidentally open and close the PR like a complete idiot

**Pull request merged**